BeamlineGeom/inc/TransportSolenoid.hh

```diff
@@ -201,27 +221,43 @@ namespace mu2e {
 
     // Cryostat
     double _rTorus;
-    double _rVac;
+    double _rVac; //approximation for all sections share the same radius
+    //by section values
+    double _ts1RVac;
+    double _ts2RVac;
+    double _ts3RVac;
+    double _ts4RVac;
+    double _ts5RVac;
+
+    bool _build_endWallD2; //whether or not to build second component of endWallD
 
     double _rIn_endWallU1; 
     double _rIn_endWallU2; 
     double _rIn_endWallD;  
+    std::vector<double> _rIn_endWallD2;
 
     double _rOut_endWallU1; 
     double _rOut_endWallU2; 
     double _rOut_endWallD;  
+    std::vector<double> _rOut_endWallD2;
+
```

Comment:
I used vectors due to SimpleConfig using vectors with defined sizes when accessing the configuration information. Is the suggestion to read in a vector and then read this into an array, or read in each value separately and load these into the array?

Reply:
To the extent my memory works, Vector was a mature class before Array was, so a lot of existing code doesn't follow this recommend because of when it was written.  That can make it tricky when you're trying to update but also trying to stay style-consistent with what is already there.  It might be a good summer project for an undergrad to look at these sorts of things...